sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1847,6 +1847,33 @@ object GpuOverrides {
         override def convertToGpu(): Expression =
           sortOrder.withNewChildren(childExprs.map(_.convertToGpu()))
       }),
+    expr[PivotFirst](
+      "PivotFirst operator",
+      ExprChecks.fullAgg(
```

Comment:
`fullAgg` indicates that it can work as a window function, and as a reduction too. I am 99% sure that Spark does not support pivot for window operations, and 100% sure that we cannot. I am less so about reductions.  Could you please update this so that only the operations that we support are included? I don't think that there is a helper function of this. They all assume that window operations are supported in Spark for the given aggregation.

Reply:
Thanks Bobby. I have updated this for only groupBy aggregates. For reduction, it fails because in the second stage of aggregation, we groupBy grouping columns and rearrange the pivotColumnValues into separate columns.